pkg/config/config.go

```diff
@@ -1,8 +1,223 @@
+// Package config defines global configuration values
 package config
 
+import (
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/puppetlabs/relay/pkg/errors"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+// OutputType may be either text or json
+type OutputType string
+
+const (
+	OutputTypeText OutputType = "text"
+	OutputTypeJSON OutputType = "json"
+)
+
+const (
+	defaultAPIDomain  = "https://api.nebula.puppet.com"
+	defaultUIDomain   = "https://nebula.puppet.com"
+	defaultWebDomain  = "https://relay.sh"
+	defaultConfigName = "config"
+	defaultConfigType = "yaml"
+)
+
 type Config struct {
-	Debug       bool   `yaml:"debug"`
-	APIHostAddr string `yaml:"apiHostAddr"`
-	CacheDir    string `yaml:"cacheDir"`
-	TokenPath   string `yaml:"tokenPath"`
+	Debug     bool
+	Out       OutputType
+	APIDomain *url.URL
+	UIDomain  *url.URL
```

Comment:
Let's make a mental note to get these from the API at some point. It knows about the UI domain already and I'm sure it can hand over the website one too.